src/client.rs

```diff
@@ -353,26 +395,22 @@ where
     ) -> Result<Client<S, T>, Error> {
         let process_id = bytes.get_i32();
         let secret_key = bytes.get_i32();
-
-        let config = get_config();
-        let transaction_mode = config.general.pool_mode == "transaction";
-        let stats = get_reporter();
-
         return Ok(Client {
             read: BufReader::new(read),
             write: write,
             addr,
             buffer: BytesMut::with_capacity(8196),
             cancel_mode: true,
-            transaction_mode: transaction_mode,
```

Comment:
cancel mode don't care about transaction_mode so it is easier to hard code a value instead of working our way backwards from the secret to the corresponding pool